lib/elixir/lib/calendar/iso.ex

```diff
@@ -71,6 +91,24 @@ defmodule Calendar.ISO do
       ]
     end
 
+  defguardp is_year(year) when year in -9999..9999
+  defguardp is_year_BCE(year) when year in -9999..0
+  defguardp is_year_CE(year) when year in 1..9999
+  defguardp is_month(month) when month in 1..12
+  defguardp is_day(day) when day in 1..31
+  defguardp is_hour(hour) when is_integer(hour) and hour >= 0
+  defguardp is_minute(minute) when is_integer(minute) and minute >= 0
+  defguardp is_second(second) when is_integer(second) and second >= 0
+
+  defguardp is_microsecond(microsecond)
```

Comment:
I would say a `tuple_size(microsecond) == 2` is enough as a check here.

Reply:
is it ok if `tuple_size/1` raises within the guard if no tuple is provided?

Reply:
Guards don't raise, otherwise we wouldn't be able to compose them with an `or` or have different function clauses that have guards.

Reply:
> Guards don't raise, otherwise we wouldn't be able to compose them with an `or` or have different function clauses that have guards.

@wojtekmach Thanks. I know that. That's why i said "within", meaning that they raise internally, but wondering if that would be considered a bad practice.
Anyway reading the Guard section I found the answer in the documention
> In the code above, we used length/1 to both check that the given thing is a list and check some properties of its length (instead of using is_list(something) and length(something) > 0).